### ISSUE TITLE
Airtango - 2019 - Get Stream URL fails

### ISSUE THREAD

**Issue by TheNorthernStar** [MRLB/kodi-airtango#1]
Hi, thanks for your kodi-addon created so far. When trying it this year it seemed to me that the URLs and logic how they are exposed have been changed.

I did some research and found the following things - hoping to help you getting it up and running again ...

1. Where to get the stream URL from?
https://www.airtango.live/api/video/<playerpage>/access?device_category_id=1
When this API is called with a proper player page it returns a file in M3U format - guess that this is already supported by Kodi - if not, it is a json format listing several streaming formats.

2. Where to get the <playerpage> from?
The homepage seems to be organised in channels now - they have a fixed name per team, e.g. for the MLP Academics it is:
https://www.airtango.live/de-int/page/mlp-academics-heidelberg 

When hovering over the images representing the scheduled games, you see a URL like
https://www.airtango.live/de-int/playerpage/76136 - this number is the one to use for <playerpage> in step 1. So parsing the pages as you did in the current version should be able to extract it from the page.

Hope that this helps in getting it up and running for this season.

**Comment by TheNorthernStar** [comment]
I just saw that the URL is corrupted (the placeholder has been suppressed ...). The correct URL with a sample id is looks like this:
https://www.airtango.live/api/video/**76136**/access?device_category_id=1
and in addition calling it via GET or POST seems to be different - with POST (empty body) you get a proper result (checked with POSTMAN), with GET the browser you might get a file download.

**Comment by MRLB** [comment]
Hi.

Mayba tomorrow during the evening I have time to check how airtango works this season.

**Comment by Zortex601** [comment]
Hi!

Did you have time to check that?

**Comment by indianahorst** [comment]
It would be very nice if you could fix this addon in the next few days! Thanks!

**Comment by MRLB** [comment]
I'll Try to take a look to the addon Monday evening. Today I am on the road :-(

**Comment by MRLB** [comment]
Hi everyone,
I have taken a look to the new api and I am surprised. The api is better as the api from last season. Much better. I have started with the new live overview. Unfortunatelly today is now livematch therefore I cannot check if it works. Hopefully I find time tomorrow to check it. Maybe you have to go into the contextmenu and click on "play from here" if the stream is not started with a "normal" click :-(

**Comment by MRLB** [comment]
Version 1.0.1 runs if you open the stream with the context menu and "play from here"

**Comment by fre2dyy** [comment]
Thanks for your work! Is it possible to add a login function to activate the 1080p streams?

**Comment by MRLB** [comment]
I have to take a look on this. Maybe next week. I added to version 1.1.0 the possiblity to start relive-streams and you do not need any more go to the contextmenu to start a stream.

**Comment by indianahorst** [comment]
> Thanks for your work! Is it possible to add a login function to activate the 1080p streams?

On my system (Raspberry Pi 4 with LibreELEC 9.2 Beta) I got the 1080p Stream automatically (without any account). I wondered about this because of the new Freemium model of Airtango (they take 2,99 EUR per month for FullHD). 

I opened the same stream in Chrome and got only the 720p stream (as expected).

MRLB: 
Indeed it would be great to have a function to switch quality (the Raspberry stuttered sometimes with 1080p, so it would be nice to be able to switch to 720p).

Another feature request: 
Would it be possible to implement a function to play the ReLive streams of the last games?

**Comment by makawi77** [comment]
Thank you, great work! But, If the games are already started, you can not find them anymore.

**Comment by MRLB** [comment]
@makawi77 Really? I can find the streams from today in "Nächste Liveevents" and for example Schwenningen - Ehingen works there

**Comment by MRLB** [comment]
@indianahorst Please go to the inputstream setting and enter there 720p. Than the addon should use the 720p stream.

I added the archive of the Pro A already. There you can find all relive-streams sorted by teamnames. 

In version 1.1.1 I added a separate ProA-Live-section and a folder with the latest relive-uploads

**Comment by makawi77** [comment]
> @makawi77 Really? I can find the streams from today in "Nächste Liveevents" and for example Schwenningen - Ehingen works there

Yes, I could also see this game next to Paderborn - Rostock in the section.  The other games, which began half an hour earlier, were no longer available at 8 pm in the section "Next Livestreams".

**Comment by MRLB** [comment]
Ok. I have to Check ist during the next gameday. Maybe there is an api available only with live streams.

**Comment by TheNorthernStar** [comment]
Hi MRLB, I tried to install it using Kodi 18.2 - and it complains about an invalid structure. 
Any idea about that? Reading the post above it seems that it works without any issue.
What version are you using?
Thanks

**Comment by TheNorthernStar** [comment]
Hi MRLB, here is an API documentation of the 2basketballiga page - a comprehensive wiki page explaining the usage of the API. Perhaps it helps in getting the list of live games? [https://wiki.2basketballbundesliga.de/index.php/API](https://wiki.2basketballbundesliga.de/index.php/API)

**Comment by makawi77** [comment]
> Ok. I have to Check ist during the next gameday. Maybe there is an api available only with live streams.

Now I could not find any more problems (Version 1.1.1)

**Comment by fre2dyy** [comment]
The addon works like a charm! Would you please integrate a login to use the 1080p stream?
